rest/config.go

```diff
@@ -790,8 +791,22 @@ func RunServer(config *ServerConfig) {
 		}()
 	}
 
+	go func() {
+		// The APIs could take a long time to start if they're blocked on view re-indexing
+		if err := waitForResponse(*config.AdminInterface); err != nil {
+			base.LogFatal("Error waiting for admin REST API: %v", err)
+		}
+		if err := waitForResponse(*config.Interface); err != nil {
+			base.LogFatal("Error waiting for public REST API: %v", err)
+		}
+
+		// Start up replicators only when the APIs are ready
+		sc.StartReplicators()
```

Comment:
I think we should make this more generic - like sc.PostStartup().  I'm fine leaving sc.StartReplicators() as-is, and have that be the only thing called by PostStartup().  I think the additional layer is merited in this case (even though we don't have any other use cases yet) to let us add some additional comments in PostStartup about how it's intended to be used, and to encourage reuse in future (instead of cluttering up this goroutine).

Reply:
Agreed. I originally had a `sc.PostStartup` method which took a callback, but in the process of removing the callback, I removed that intermediate step. I'll re-add that and have an explicit call to `sc.StartReplicators()`